### PR TITLE
(FULL) Easybuild update integration

### PR DESCRIPTION
**Pull request by NeutryFD**
Add  new options  --complete-dependencies  for R packages 
- Update version and checksum from  CRAN database to easyconfig file 
- Write  dependecies tree of extension into easyconfig file with the correct order for the installation
-  Create a .bk file of the original easyconfig file

### CODE REVIEW

**Comment by macheitor** [comment]
Improved NeutryFD's version of complete_dependencies. 
There are several TODO's pending to be solved by someone with more experience in EasyBuild

**Comment by macheitor** [comment]
Changed flag to "--complete-exts-list" which is literally what we are doing. 
We are not updating the version of the packages already present on the exts_list. 
This would be another feature, also is really easy to do this on the current code.

It is working now for R, R-Bioconductor and Python

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -4835,3 +4845,1025 @@ def make_checksum_lines(checksums, indent_level):
             ectxt = regex.sub('\n'.join(exts_list_lines), ectxt)
 
         write_file(ec['spec'], ectxt)
+
+
+def exts_list_utils(ecs, complete=False, update=False):
```
Comment (dagonzalezfo) [inline easybuild/framework/easyblock.py]: This exts_utils can be moved to tools.py

File: easybuild/framework/easyblock.py [hunk 2]
```diff
@@ -4835,3 +4845,1025 @@ def make_checksum_lines(checksums, indent_level):
...
+                # Aesthetic print
+                print_msg('')
+
+                return self.update_exts_list
+
+            except Exception as e:
+                print_error(f"Failed to get complete exts list: {e}", log=_log)
+                return None
+        
+    class PythonExtension:
```
Comment (dagonzalezfo) [inline easybuild/framework/easyblock.py]: Check available EB clases to avoid clashes/redundancies

**Review by dagonzalezfo — COMMENTED** [review]
Do changes as proposed in meet session:
Use available EB clases if available and add funcions if needed overcharge it by adding new funcionalities.

Move ext related tooling (updates and other magic) into tools.py.

**Comment by dagonzalezfo** [comment]
By exploring with Victor where is the best way to add our deployment, we decide to go by extending `easybuild/framework/extension.py` by adding properties such as : 
- `update_deps`
- `complete_deps`
- and provide a way to get `required_deps`

https://github.com/easybuilders/easybuild-framework/blob/e0e04fb773a5cee26669211f563696da94455cfd/easybuild/framework/extension.py#L238

**Pull request closed without merging**